sql/server.go

```diff
@@ -153,37 +153,45 @@ func (p parameters) Arg(i int) (parser.Datum, bool) {
 
 // exec executes the request. Any error encountered is returned; it is
 // the caller's responsibility to update the response.
-func (s *Server) exec(req driver.Request) (driver.Response, error) {
-	var resp driver.Response
+func (s *Server) exec(req driver.Request) (resp driver.Response, err error) {
 
 	// Pick up current session state.
 	// The request user is validated in ServeHTTP. Even in insecure mode,
 	// it is guaranteed not to be empty.
-	planner := planner{user: req.GetUser()}
+	planMaker := planner{user: req.GetUser()}
+	defer func(p *planner) {
+		// Update session state.
+		var marshallErr error
+		if resp.Session, marshallErr = gogoproto.Marshal(&p.session); marshallErr != nil {
+			err = marshallErr
+		}
+	}(&planMaker)
```

Comment:
done. Turns out that does work. I thought the context works like function parameters, but it actually can change under the covers with defer using the freshest value of planMaker. Thanks